app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,44 @@
+class PostsController < ApplicationController
+  def create
+    @post = current_user.posts.new(post_params)
+    @post.author_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```

Comment:
- [ ] Well done but you should use CanCanCan for Post authorization:
`  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
`